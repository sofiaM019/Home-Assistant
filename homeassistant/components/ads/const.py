"""Support for Automation Device Specification (ADS)."""

from __future__ import annotations

from enum import StrEnum
from typing import TYPE_CHECKING

from homeassistant.util.hass_dict import HassKey

if TYPE_CHECKING:
    from .hub import AdsHub

DOMAIN = "ads"

DATA_ADS: HassKey[AdsHub] = HassKey(DOMAIN)
STATE_KEY_STATE = "state"

SERVICE_WRITE_DATA_BY_NAME = "write_data_by_name"

CONF_ADS_VAR = "adsvar"
CONF_ADS_FACTOR = "factor"
CONF_ADS_TYPE = "adstype"
CONF_ADS_VALUE = "value"


class AdsType(StrEnum):
    """Supported Types."""

    BOOL = "bool"
    BYTE = "byte"
    INT = "int"
    UINT = "uint"
    SINT = "sint"
    USINT = "usint"
    DINT = "dint"
    UDINT = "udint"
    WORD = "word"
    DWORD = "dword"
    LREAL = "lreal"
    REAL = "real"
    STRING = "string"
    TIME = "time"
    DATE = "date"
    DATE_AND_TIME = "dt"
    TOD = "tod"
