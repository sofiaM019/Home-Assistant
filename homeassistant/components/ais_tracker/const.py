"""AIS tracker constants."""

DOMAIN = "ais_tracker"

CONF_MMSIS = "mmsis"
DEFAULT_PORT = 12346
