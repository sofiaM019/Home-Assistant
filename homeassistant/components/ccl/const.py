"""Constants for the CCL Electronics integration."""

DOMAIN = "ccl"
