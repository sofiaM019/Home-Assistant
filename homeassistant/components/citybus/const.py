"""CityBus constants"""

DOMAIN = "citybus"

CONF_ROUTE = "route"
CONF_DIRECTION = "direction"
CONF_STOP = "stop"
