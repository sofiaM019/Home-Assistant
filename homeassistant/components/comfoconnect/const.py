"""Constants for the ComfoConnect integration."""

DOMAIN = "comfoconnect"

CONF_USER_AGENT = "user_agent"
DEFAULT_PIN = 0
DEFAULT_TOKEN = "00000000000000000000000000000001"
DEFAULT_USER_AGENT = "Home Assistant"
