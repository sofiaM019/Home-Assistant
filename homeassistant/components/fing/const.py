"""Const for the Fing integration."""

AGENT_IP = "agent_ip"
AGENT_PORT = "agent_port"
AGENT_KEY = "agent_key"
DOMAIN = "fing"
