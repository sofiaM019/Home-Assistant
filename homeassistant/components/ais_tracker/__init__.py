"""AIS tracker."""

import logging

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant

from .coordinator import AisTrackerConfigEntry, AisTrackerCoordinator

LOGGER = logging.getLogger(__name__)

PLATFORMS = [
    Platform.DEVICE_TRACKER,
    Platform.SENSOR,
]


async def async_setup_entry(hass: HomeAssistant, entry: AisTrackerConfigEntry) -> bool:
    """Set up config entry."""

    coordinator = AisTrackerCoordinator(hass)
    await coordinator.async_setup()

    entry.runtime_data = coordinator

    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, entry: AisTrackerConfigEntry) -> bool:
    """Unload a config entry."""
    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
