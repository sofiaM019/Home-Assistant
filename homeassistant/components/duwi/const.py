"""Constants for the DUWI integration."""

from enum import StrEnum
import logging

from homeassistant.const import Platform

_LOGGER = logging.getLogger(__package__)

MANUFACTURER = "duwi"

DOMAIN = "duwi"

APP_VERSION = "0.1.1"

CLIENT_VERSION = "0.1.1"

CLIENT_MODEL = "homeassistant"

DUWI_DISCOVERY_NEW = "duwi_discovery_new"
DUWI_SCENE_UPDATE = "duwi_scene_update"
DUWI_HA_SIGNAL_UPDATE_ENTITY = "duwi_entry_update"
DUWI_HA_ACCESS_TOKEN = "duwi_access_token"

# API keys
CLIENT = "client"
ADDRESS = "address"
WS_ADDRESS = "ws_address"
APP_KEY = "app_key"
APP_SECRET = "app_secret"
ACCESS_TOKEN = "access_token"
REFRESH_TOKEN = "refresh_token"
PHONE = "phone"
PASSWORD = "password"
HOUSE_NO = "house_no"
HOUSE_NAME = "house_name"
HOUSE_KEY = "house_key"
CONF_TOKEN_INFO = "token_info"

# List of platforms that support config entry
SUPPORTED_PLATFORMS = [
    Platform.SWITCH,
]


class DPCode(StrEnum):
    """Data Point Codes used by Duwi."""

    # button
    BUTTON = "button"

    # switch
    SWITCH = "switch"

    # light
    ON_OFF = "on_off"
    LIGHT = "light"
    COLOR_TEMP = "color_temp"
    COLOR = "color"
    RGB = "rgb"
    RGBW = "rgbw"
    RGBCW = "rgbcw"

    # cover
    ROLLER = "roller"
    SHUTTER = "shutter"
    SHUTTER_2 = "shutter-2"

    # music
    HUA_ERSI_MUSIC = "hua_ersi_music"
    XIANG_WANG_MUSIC_S7_MINI_3S = "xiangwang_music_s7_mini_3s"
    XIANG_WANG_MUSIC_S8 = "xiangwang_music_s8"
    SHENG_BI_KE_MUSIC = "sheng_bike_music"
    BO_SHENG_MUSIC = "bo_sheng_music"
    YOU_DA_MUSIC = "youda_music"

    # sensor
    TEMP = "temp"
    HUMIDITY = "humidity"
    HCHO = "hcho"
    PM25 = "pm25"
    CO2 = "co2"
    BRIGHT = "bright"
    IQA = "iaq"
    HUMAN = "human"
    TRIGGER = "trigger"
    MOISTURE = "moisture"
    SMOKE = "smoke"
    MOVING = "moving"
    GAS = "gas"
    DOOR = "door"
    OPENING = "opening"
    CO = "co"
    TVOC = "tvoc"
    PM10 = "pm10"

    # climate
    AC = "ac"
    FA = "fa"
    FH = "fh"
    HP = "hp"
    TC = "tc"
