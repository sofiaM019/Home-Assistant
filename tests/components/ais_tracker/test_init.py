"""Tests for AIS tracker config flow."""

from collections.abc import Generator
from unittest.mock import MagicMock, patch

from pyais.messages import NMEAMessage
from pyais.stream import SocketStream
from syrupy import SnapshotAssertion

from homeassistant.components.ais_tracker.const import DOMAIN
from homeassistant.core import HomeAssistant

from .const import MOCK_USER_DATA

from tests.common import MockConfigEntry


class MockUDPReceiver(SocketStream):
    """Mock UDP receiver."""

    def __init__(self, *args, **kwargs) -> None:
        """Initialize the mock."""
        self.messages = [
            NMEAMessage(b"!AIVDM,1,1,,B,139f1wPP0CPwM>JM<3eiK?wf25KT,0*60"),
            NMEAMessage(
                b"!AIVDM,2,1,4,A,539f1w`00000@?GK;D1<<Phu=<H4DQ8D0000000t20=72t0Ht1P00000,0*6D"
                b"!AIVDM,2,2,4,A,000000000000000,2*20"
            ),
        ]
        socket = MagicMock()
        super().__init__(socket, preprocessor=None)

    def __iter__(self) -> Generator[NMEAMessage, None, None]:
        """Iterate over messages."""
        while True:
            yield from [self.messages.pop()]


async def test_init(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test init of AIS tracker component."""
    mock_config = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
    mock_config.add_to_hass(hass)

    with patch(
        "homeassistant.components.ais_tracker.coordinator.UDPReceiver",
        new=MockUDPReceiver,
    ):
        await hass.config_entries.async_setup(mock_config.entry_id)
        await hass.async_block_till_done()

    states = hass.states.async_all()
    assert len(states) == 8

    for state in states:
        assert state == snapshot(name=state.entity_id)
