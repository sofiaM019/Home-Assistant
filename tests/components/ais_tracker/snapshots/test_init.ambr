# serializer version: 1
# name: test_init[device_tracker.211517950]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': '211517950',
      'gps_accuracy': 0,
      'latitude': 51.010438,
      'longitude': 13.862315,
      'source_type': <SourceType.GPS: 'gps'>,
    }),
    'context': <ANY>,
    'entity_id': 'device_tracker.211517950',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'not_home',
  })
# ---
# name: test_init[sensor.211517950_course]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': '211517950 Course',
      'unit_of_measurement': '°',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.211517950_course',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '36.4',
  })
# ---
# name: test_init[sensor.211517950_heading]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': '211517950 Heading',
      'unit_of_measurement': '°',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.211517950_heading',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_init[sensor.211517950_maneuver]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'enum',
      'friendly_name': '211517950 Maneuver',
      'options': list([
        'NotAvailable',
        'NoSpecialManeuver',
        'SpecialManeuver',
        'UNDEFINED',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'sensor.211517950_maneuver',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'NotAvailable',
  })
# ---
# name: test_init[sensor.211517950_speed]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'speed',
      'friendly_name': '211517950 Speed',
      'unit_of_measurement': <UnitOfSpeed.KNOTS: 'kn'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.211517950_speed',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '0.19',
  })
# ---
# name: test_init[sensor.211517950_status]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'enum',
      'friendly_name': '211517950 Status',
      'options': list([
        'UnderWayUsingEngine',
        'AtAnchor',
        'NotUnderCommand',
        'RestrictedManoeuverability',
        'ConstrainedByHerDraught',
        'Moored',
        'Aground',
        'EngagedInFishing',
        'UnderWaySailing',
        'ReservedFutureAmendmentHSC',
        'ReservedFutureAmendmentWIG',
        'PowerDrivenVesselTowingAstern',
        'PowerDrivenVesselPushingAhead',
        'ReservedFutureUse',
        'AISSARTActive',
        'Undefined',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'sensor.211517950_status',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'UnderWayUsingEngine',
  })
# ---
# name: test_init[sensor.211517950_turn]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': '211517950 Turn',
      'unit_of_measurement': '°/min',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.211517950_turn',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_init[zone.home]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'editable': True,
      'friendly_name': 'test home',
      'icon': 'mdi:home',
      'latitude': 32.87336,
      'longitude': -117.22743,
      'passive': False,
      'persons': list([
      ]),
      'radius': 100,
    }),
    'context': <ANY>,
    'entity_id': 'zone.home',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '0',
  })
# ---
